src/routes/login.tsx

```diff
@@ -87,13 +87,17 @@ export function LoginForm() {
   }, [currentRole, currentType]);
 
   useLayoutEffect(() => {
-    if (isAuthenticated && search?.redirect) {
-      router.history.push(search.redirect);
-    } else if (isAdmin) {
-      router.history.push('/admin/xpub');
-    } else if (isUser) {
-      router.history.push('/user/access-keys');
+    if (!isAuthenticated) {
+      return;
     }
+
+    let redirect = search?.redirect?.includes('user') ? search?.redirect : '/user/access-keys';
+
+    if (isAdmin) {
+      redirect = search?.redirect?.includes('admin') ? search?.redirect : '/admin/xpub';
+    }
```

Comment:
```suggestion
    let redirect = ""

    if (isAdmin) {
      redirect = search?.redirect?.includes('admin') ? search?.redirect : '/admin/xpub';
    } else {
      redirect = search?.redirect?.includes('user') ? search?.redirect : '/user/access-keys';
    }
```

Reply:
I wanted to omit the else statement but sure. If it's more readable that way I'll change it :D